src/mscorlib/shared/System/Runtime/CompilerServices/AsyncMethodBuilder.cs

```diff
@@ -14,6 +14,7 @@ internal static partial class AsyncMethodBuilder
         /// <typeparam name="TStateMachine">Specifies the type of the state machine.</typeparam>
         /// <param name="stateMachine">The state machine instance, passed by reference.</param>
         [DebuggerStepThrough]
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
```

Comment:
Why?

Reply:
> the original motivation for suggesting removing the try/finally was to enable inlining. I seriously doubt that in its current form the method will inline

Just following through to logical conclusion before going back
```
Total bytes of diff: -3214 (-0.09% of base)
    diff is an improvement.

Total byte diff includes 0 bytes from reconciling methods
        Base had    0 unique methods,        0 unique bytes
        Diff had    0 unique methods,        0 unique bytes

Top file improvements by size (bytes):
       -3214 : System.Private.CoreLib.dasm (-0.09% of base)

1 total files with size differences (1 improved, 0 regressed), 0 unchanged.

Top method regessions by size (bytes):
         102 : System.Private.CoreLib.dasm - AsyncVoidMethodBuilder:Start(byref):this (2 methods)
         102 : System.Private.CoreLib.dasm - AsyncTaskMethodBuilder:Start(byref):this
         102 : System.Private.CoreLib.dasm - AsyncTaskMethodBuilder`1:Start(byref):this (2 methods)
         102 : System.Private.CoreLib.dasm - AsyncValueTaskMethodBuilder`1:Start(byref):this
          98 : System.Private.CoreLib.dasm - StreamWriter:WriteAsyncInternal(ref,ref,ref,int,int,ref,bool,bool):ref
          98 : System.Private.CoreLib.dasm - SemaphoreSlim:WaitUntilCountOrTimeoutAsync(ref,int,struct):ref:this
          94 : System.Private.CoreLib.dasm - CommonlyUsedGenericInstantiations:AsyncHelper() (2 methods)
          94 : System.Private.CoreLib.dasm - CommonlyUsedGenericInstantiations:AsyncHelper2():ref (2 methods)
          94 : System.Private.CoreLib.dasm - CommonlyUsedGenericInstantiations:AsyncHelper3():ref
          94 : System.Private.CoreLib.dasm - Stream:<ReadAsync>g__FinishReadAsync|44_0(ref,ref,struct):struct
          94 : System.Private.CoreLib.dasm - StreamReader:ReadAsyncInternal(struct,struct):struct:this
          94 : System.Private.CoreLib.dasm - TextReader:ReadBlockAsyncInternal(struct,struct):struct:this
          92 : System.Private.CoreLib.dasm - Stream:CopyToAsyncInternal(ref,int,struct):ref:this
          92 : System.Private.CoreLib.dasm - Stream:<WriteAsync>g__FinishWriteAsync|57_0(ref,ref):ref
          92 : System.Private.CoreLib.dasm - FileStream:AsyncModeCopyToAsync(ref,int,struct):ref:this
          92 : System.Private.CoreLib.dasm - StreamReader:ReadLineAsyncInternal():ref:this
          92 : System.Private.CoreLib.dasm - StreamReader:ReadToEndAsyncInternal():ref:this
          92 : System.Private.CoreLib.dasm - StreamReader:ReadBufferAsync():ref:this
          92 : System.Private.CoreLib.dasm - StreamWriter:WriteAsyncInternal(ref,ushort,ref,int,int,ref,bool,bool):ref
          92 : System.Private.CoreLib.dasm - StreamWriter:WriteAsyncInternal(ref,struct,ref,int,int,ref,bool,bool,struct):ref
          92 : System.Private.CoreLib.dasm - StreamWriter:FlushAsyncInternal(ref,bool,bool,ref,int,bool,ref,ref,ref,ref,struct):ref
          92 : System.Private.CoreLib.dasm - TextReader:ReadToEndAsync():ref:this

Top method improvements by size (bytes):
       -5302 : System.Private.CoreLib.dasm - AsyncMethodBuilder:Start(byref) (21/3 methods)

23 total methods with size differences (1 improved, 22 regressed), 17132 unchanged.
```